RecursiveExtractor/Extractors/SevenZipExtractor.cs

```diff
@@ -97,11 +106,29 @@
                         try
                         {
                             sevenZipArchive = SevenZipArchive.Open(fileEntry.Content, new SharpCompress.Readers.ReaderOptions() { Password = password });
-                            if (sevenZipArchive.TotalUncompressSize > 0)
+                            // When filenames are encrypted we can't access the size of individual files
+                            // But if we can accesss the total uncompressed size we have the right password
+                            try
+                            {
+                                var entry = sevenZipArchive.Entries.FirstOrDefault(x => x.IsEncrypted && x.Size > 0);
+                                // When filenames are plain, we can access the properties, so the previous statement won't throw
+                                // Instead we need to check that we can actually read from the stream
+                                using var entryStream = entry?.OpenEntryStream();
+                                var bytes = new byte[1];
+                                if ((entryStream?.Read(bytes, 0, 1) ?? 0) == 0)
+                                {
+                                    Logger.Trace(Extractor.FAILED_PASSWORD_STRING, fileEntry.FullPath, ArchiveFileType.P7ZIP);
```

Comment:
Is this meant to be a failed password error string? MIght be worth updating the name to FAILED_PASSWORD_ERROR_STRING to avoid future confusion during code analysis.